R/fit_best.R

```diff
@@ -0,0 +1,147 @@
+#' Fit a model to the numerically optimal configuration
+#'
+#' `fit_best()` takes the results from model tuning and fits it to the training
+#' set using tuning parameters associated with the best performance.
+#'
+#' @param x The results of class `tune_results` (coming from functions such as
+#' [tune_grid()], [tune_bayes()], etc). The control option
+#' [`save_workflow = TRUE`][tune::control_grid] should have been used.
+#' @param metric A character string (or `NULL`) for which metric to optimize. If
+#' `NULL`, the first metric is used.
+#' @param parameters An optional 1-row tibble of tuning parameter settings, with
+#' a column for each tuning parameter. This tibble should have columns for each
+#' tuning parameter identifier (e.g. `"my_param"` if `tune("my_param")` was used).
+#' If `NULL`, this argument will be set to
+#' [`select_best(metric)`][tune::select_best.tune_results].
+#' @param verbose A logical for printing logging.
+#' @param ... Not currently used.
+#' @details
+#' This function is a shortcut for the manual steps of:
+#'
+#' \preformatted{
+#'   best_param <- select_best(tune_results, metric) # or other `select_*()`
+#'   wflow <- finalize_workflow(wflow, best_param)  # or just `finalize_model()`
+#'   wflow_fit <- fit(wflow, data_set)
+#' }
+#'
+#' The data used for the fit are taken from the `splits` column. If the split
+#' column was from a validation split, the combined training and validation sets
+#' are used.
+#'
+#' In comparison to [last_fit()], that function requires a finalized model, fits
+#' the model on the data set by [rsample::initial_split()], and computes
```

Comment:
I think the phrasing

> "fits the model on the data set by [rsample::initial_split()]" 

could use some clarification.

Reply:
How about "the training set defined by" the initial split?

Reply:
That sounds great!